sysutils/pfSense-pkg-node_exporter/files/usr/local/pkg/node_exporter.inc

```diff
@@ -0,0 +1,133 @@
+<?php
+/*
+ * node_exporter.inc
+ *
+ * part of pfSense (https://www.pfsense.org)
+ * Copyright (c) 2009-2015 Rubicon Communications, LLC (Netgate)
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+require_once("config.inc");
+require_once("functions.inc");
+require_once("util.inc");
+require_once("service-utils.inc");
+
+const NODE_EXPORTER_CONFIG_FILE = "/usr/local/etc/rc.conf.d/node_exporter";
+const NODE_EXPORTER_RCFILE = "/usr/local/etc/rc.d/node_exporter.sh";
+const NODE_EXPORTER_SHARE_DIR = "/usr/local/share/pfSense-pkg-node_exporter";
+const NODE_EXPORTER_RCFILE_UPSTREAM = NODE_EXPORTER_SHARE_DIR . "/node_exporter.sh";
+const NODE_EXPORTER_DEFAULT_COLLECTORS = "boottime,cpu,exec,filesystem,loadavg,meminfo,netdev,textfile,time";
+const NODE_EXPORTER_TEXTFILE_CRONTASK = "/usr/bin/nice -n20 /usr/local/bin/python2.7 ".NODE_EXPORTER_SHARE_DIR."/interface-collector.py";
+
+function node_exporter_write_config($node_exporter_config) {
+	$txt = "# This file is generated by the pfSense node_exporter package.\n";
+	$txt .= "# Do not edit this file, it will be overwritten automatically.\n";
+	$txt .= "\n";
+	$txt .= "node_exporter_enable=\"YES\"\n"; // If it had been NO, this function isn't called, so fine to hardcode
+
+	$listen_addr = get_interface_ip($node_exporter_config["listen_iface"]);
+	$listen_port = $node_exporter_config["listen_port"] ?: "9100";
+	$txt .= "node_exporter_listen_address=\"" . $listen_addr . ":" . $listen_port . "\"\n";
+
+	$collectorTxt = "";
+	foreach (explode(",", $node_exporter_config["collectors"]) as $collector) {
+		$collectorTxt .= " --collector.".$collector;
+	}
+
+	$txt .= "node_exporter_args=\"". $collectorTxt ." " . $node_exporter_config["extra_flags"] . "\"\n";
```

Comment:
Similar to the previous note, simplify this a little:
```
$txt .= "node_exporter_args=\"{$collectorTxt} {$node_exporter_config['extra_flags']}\"\n";
```
(Note the change to single quotes around `extra_flags`)